### PR TITLE
Support SSL in node-to-node communication

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mfussenegger** [comment]
@mkleen Could you have a first high level view over this?

Docs and HBA integration is missing, but forced SSL and dual mode SSL itself should be good so far.

**Comment by mkleen** [comment]
Looks pretty good on a first view. Do you think it would make sense to maybe include the ssl status as part of `sys.node` to make it accessible ?

**Comment by mfussenegger** [comment]
> Looks pretty good on a first view. Do you think it would make sense to include the ssl status such as 'ssl-active' as part of `sys.node` to make it accessible ?

Exposing the node setting would be possible I guess, but exposing the state of each connection might be too much, as there are point-to-point connections between all the nodes, so it would blow up the output a lot.

**Comment by mfussenegger** [comment]
Changed the base to a feature branch, so that this can already be reviewed and merged if it's good.

File: server/src/main/java/io/crate/protocols/ssl/SslContextProvider.java [hunk 1]
```diff
@@ -85,19 +98,65 @@ public SslContext getSslContext() {
...
+            sslContext = serverContext();
             LOGGER.info("SSL configuration is reloaded.");
         }
     }
 
-    private SslContext buildSslContext() {
-        var keystorePath = SslSettings.SSL_KEYSTORE_FILEPATH.get(settings);
-        var keystorePass = SslSettings.SSL_KEYSTORE_PASSWORD.get(settings).toCharArray();
-        var keystoreKeyPass = SslSettings.SSL_KEYSTORE_KEY_PASSWORD.get(settings).toCharArray();
+    public SSLContext jdkSSLContext() throws Exception {
```
Comment (seut) [inline server/src/main/java/io/crate/protocols/ssl/SslContextProvider.java:106]: maybe add a comment that this is used for testing?

**Review by seut — APPROVED** [review]
👍 very nice. added a comment suggestion

File: server/src/main/java/io/crate/protocols/ssl/SslContextProvider.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/protocols/ssl/SslContextProvider.java:106]: Added a `VisibleForTesting` annotation

**Pull request merged**